src/inc/til/small_vector.h

```diff
@@ -357,57 +359,39 @@ namespace til
         small_vector(const small_vector& other) :
             small_vector{}
         {
-            operator=(other);
+            _copy_assign(other);
         }
 
         // NOTE: If an exception is thrown while copying, the vector is left empty.
         small_vector& operator=(const small_vector& other)
         {
-            clear();
-            reserve(other._size);
-
-            std::uninitialized_copy(other.begin(), other.end(), _uninitialized_begin());
-            _size = other._size;

```

Comment:
These 3 lines are now moved 1:1 into `_copy_assign`. This allows us to call `_copy_assign` without a prior `clear` in the constructor, because a newly constructed vector can't possibly have anything that needs to be cleared.

This makes the diff large, but actually barely anything changed.